src/Domain/Creators/Interval.cpp

```diff
@@ -49,12 +52,36 @@ Interval::Interval(
 }
 
 Domain<1> Interval::create_domain() const noexcept {
+  std::vector<DirectionMap<
+      1, std::unique_ptr<domain::BoundaryConditions::BoundaryCondition>>>
+      boundary_conditions_all_blocks{};
+  if (lower_boundary_condition_ != nullptr) {
+    if (upper_boundary_condition_ == nullptr)  {
```

Comment:
Question to check my understanding of the interface: what does the user do in the input file to get a nullptr here? Is it by leaving the option blank, like this?
```
BoundaryConditions:
  LowerBoundary:
  UpperBoundary:
```

Reply:
This is an annoying one that I haven't settled yet... I think the best option is to remove the `IsPeriodicIn` stuff completely and add a system-agnostic `Periodic` condition that is checked.